flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/CheckpointBarrierUnaligner.java

```diff
@@ -117,17 +117,6 @@
 		threadSafeUnaligner = new ThreadSafeUnaligner(totalNumChannels,	checkNotNull(channelStateWriter), this);
 	}
 
-	@Override
-	public void releaseBlocksAndResetBarriers() {
-		if (isCheckpointPending()) {
-			// make sure no additional data is persisted
-			Arrays.fill(hasInflightBuffers, false);
-			// the next barrier that comes must assume it is the first
-			numBarrierConsumed = 0;
-		}
-		threadSafeUnaligner.resetReceivedBarriers(currentConsumedCheckpointId);
-	}
-
```

Comment:
Are you sure this is a safe delete? `AlternatingCheckpointBarrierHandler` is for example using this method to clean up state of the previous handler before switching to another.

Reply:
I also found the  issues caused by `AlternatingCheckpointBarrierHandler` via the failed unit tests. I will consider how to refactor it to make `AlternatingCheckpointBarrierHandler` go through.

Reply:
If we change how this method is used, I'd consider it more than a hotfix. Then I'd also like to see the reason for the removal in the commit message (I don't understand it from this commit's perspective).